lib/Star/Kata/KataDomain/KataService.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * This file is part of the phpkata project.
+ *
+ * (c) Yannick Voyer (http://github.com/yvoyer)
+ */
+
+namespace Star\Kata\KataDomain;
+
+use Star\Kata\Model\KataCollection;
+use Star\Kata\Model\KataRepository;
+
+/**
+ * Class KataService
+ *
+ * @author  Yannick Voyer (http://github.com/yvoyer)
+ *
+ * @package Star\Kata\KataDomain
+ */
+final class KataService
+{
+    const CLASS_NAME = __CLASS__;
+
+    /**
+     * @var KataRepository
+     */
+    private $collection;
+
+    /**
+     * @param KataRepository $repository
+     */
+    public function __construct(KataRepository $repository)
+    {
+        $this->collection = $repository;
```

Comment:
strange variable naming.. `$this->katas = $katas` would probably be more clear. calling it `collection` or `$repository` is the equivalent of calling a variable `$nameString` or `$dataArray`.. should name your variables after data types.